### PR TITLE
Merge `package:mockito`

### PR DESCRIPTION
**Pull request by mosuem**
- [x] Move and fix workflow files, labeler.yaml, and badges in the README.md
- [x] Rev the version of the package, so that pub.dev points to the correct site
- [x] Add a line to the changelog:
```
* Move to `dart-lang/test` monorepo.
```

- [x] Add the package to the top-level readme of the monorepo:
```
| [mockito](pkgs/mockito/) | A mock framework inspired by Mockito with APIs for Fakes, Mocks, behavior verification, and stubbing. | [![pub package](https://img.shields.io/pub/v/mockito.svg)](https://pub.dev/packages/mockito) |
```

- [ ] **Important!** Merge the PR with 'Create a merge commit' (enabling then disabling the `Allow merge commits` admin setting)
- [x] Update the auto-publishing settings on https://pub.dev/packages/mockito/admin
- [x] Add the following text to https://github.com/dart-lang/mockito/:'

```
> [!IMPORTANT]  
> This repo has moved to https://github.com/dart-lang/test/tree/master/pkgs/mockito
```

- [ ] Publish using the autopublish workflow
- [ ] Push tags to GitHub using `git tag --list 'mockito*' | xargs git push origin`
- [ ] Close open PRs in dart-lang/mockito with the following message:

```
Closing as the [dart-lang/mockito](https://github.com/dart-lang/mockito) repository is merged into the [dart-lang/test](https://github.com/dart-lang/test) monorepo. Please re-open this PR there!
```
      
- [ ] Transfer issues by running `dart run pkgs/repo_manage/bin/report.dart transfer-issues --source-repo dart-lang/mockito --target-repo dart-lang/test --add-label package:mockito --apply-changes`
- [ ] Archive https://github.com/dart-lang/mockito/
- [ ] Import to g3 - see internal cl/705463525

---

- [x] I’ve reviewed the contributor guide and applied the relevant portions to this PR.

<details>
  <summary>Contribution guidelines:</summary><br>

- See our [contributor guide](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md) for general expectations for PRs.
- Larger or significant changes should be discussed in an issue before creating a PR.
- Contributions to our repos should follow the [Dart style guide](https://dart.dev/guides/language/effective-dart) and use `dart format`.
- Most changes should add an entry to the changelog and may need to [rev the pubspec package version](https://github.com/dart-lang/sdk/blob/main/docs/External-Package-Maintenance.md#making-a-change).
- Changes to packages require [corresponding tests](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md#Testing).

Note that many Dart repos have a weekly cadence for reviewing PRs - please allow for some latency before initial review feedback.
</details>

### CODE REVIEW

File: .github/workflows/mockito.yaml [hunk 1]
```diff
@@ -0,0 +1,109 @@
+name: package:mockito
```
Comment (jakemac53) [inline .github/workflows/mockito.yaml:1]: Can we migrated this to use package:mono_repo (add a `mono_repo.yaml`)?

It won't be a good situation to have multiple packages in the same repo managing their CI in different ways.

Comment (natebosch) [reply .github/workflows/mockito.yaml:1]: As long as mono_repo doesn't break in this situation, I'm not very worried about mixing the types of workflows we use in a single repo.

What types of problems are you worried might come up?

Comment (mosuem) [reply .github/workflows/mockito.yaml:1]: If we switch, I would prefer switching to `mono_repo` in a later PR. I have no experience rewriting CI files into the `mono_pkg.yaml`, and would prefer landing this merge rather quickly before anyone updates the original repository.

(To be had in a separate discussion, but I am not a big fan of `package:mono_repo`. It introduces complexity in a new file format with `mono_pkg.yaml`, generates large hard to read `dart.yml` files, and the status checks at the end of PRs do not look nice. In my experience, the CI files need little refactorings, so I don't see the advantage of generating them.)

Comment (jakemac53) [reply .github/workflows/mockito.yaml:1]: > What types of problems are you worried might come up?

It is just going to be a maintenance headache having to deal with two different systems. I don't want to manage manual github workflows.

Comment (jakemac53) [reply .github/workflows/mockito.yaml:1]: I am fine with doing this in a followup

**Comment by mosuem** [comment]
@srawlins, the `analyze` job is failing, which seems to already have been the case in the [other](https://github.com/dart-lang/mockito/actions/runs/11278396977/job/31366748910) repository. Is this a bug which needs to be investigated, or is this a known behavior?

**Comment by srawlins** [comment]
I don't see the pubspec, but I imagine the deps just need to be bumped. Or mockito in google3 might be using unpublished elements from analyzer.

**Comment by mosuem** [comment]
> I don't see the pubspec, but I imagine the deps just need to be bumped. Or mockito in google3 might be using unpublished elements from analyzer.

It seems like stable Dart does not yet support some features. I would advise to move this package with the failing CI, and fix the issue in a follow up. IIUC, fixing would require depending on a dev version of Dart, or removing the use of these features.

**Comment by jakemac53** [comment]
I looked a bit at the issue tracker for this package and I am pretty concerned about merging it into the test repo.

This package is active enough I think it should stay in its own repo. I am not able or willing to take ownership of it and don't want to be on the hook for fielding the issues, but would feel obligated to if it was in this repo.

**Comment by srawlins** [comment]
I don't have any feelings about where it lives. I have not been a good owner, but I won't ask anyone else to.

**Comment by devoncarew** [comment]
mockito is pretty up in the air currently (should we own it? do we have the time to maintain and move it forward?). Let's hold off on moving this to a monorepo until we have more clarity on the medium term / long term plans.

**Pull request closed without merging**